src/components/pages/PeoplePage.tsx

```diff
@@ -0,0 +1,119 @@
+import { FC, useState, useEffect } from 'react';
+import { Link, useParams } from 'react-router-dom';
+import { Person } from '../../types';
+import { Loader } from '../Loader';
+import { getPeople } from '../../api';
+import cn from 'classnames';
+
+export const PeoplePage: FC = () => {
+  const [peoples, setPeoples] = useState<Person[]>([]);
+  const [error, setError] = useState(false);
+  const [loading, setLoading] = useState(false);
+
+  const { personSlug } = useParams();
+
+  const personInfo = ['Name', 'Sex', 'Born', 'Died', 'Mother', 'Father'];
+
+  const personName = (name: string | null) => {
+    return peoples.find(person => person.name === name);
```

Comment:
The `personName` function is not optimally implemented. Every time you call this function, it iterates over the entire `peoples` array to find a match. If the `peoples` array is large, this can lead to performance issues. You could consider creating a map (Object or Map) where the keys are the names of the people, and the values are the people themselves. This would allow you to find a person by name in constant time.